thewalrus/fock_gradients.py

```diff
@@ -23,36 +23,72 @@
 .. autosummary::
     :toctree: api
 
-    Xgate
     Dgate
     Sgate
     Rgate
     Kgate
     S2gate
     BSgate
-    Sgate_real
-    S2gate_real
-    BSgate_real
 
 """
 import numpy as np
 
 from numba import jit
 
 
-from thewalrus.libwalrus import (
-    interferometer,
-    squeezing,
-    displacement,
-    interferometer_real,
-    displacement_real,
-    squeezing_real,
-    two_mode_squeezing,
-    two_mode_squeezing_real,
-)
+@jit(nopython=True)
+def displacement_rec(alpha, cutoff):  # pragma: no cover
+    r"""Calculate the matrix elements of the real or complex displacement gate using a recursion relation.
 
+    Args:
+        alpha (float or complex): value of the displacement.
+        cutoff (int): Fock ladder cutoff
+
+    Returns:
+        (array): matrix representing the displacement operation.
+    """
+    D = np.zeros((cutoff, cutoff), dtype=np.complex128)
+    y = np.array([alpha, -np.conj(alpha)])
+    sqns = np.sqrt(np.arange(cutoff))
+    D[0, 0] = np.exp(-0.5 * np.abs(y[0]) ** 2)
+    D[1, 0] = y[0] * D[0, 0]
+    for m in range(2, cutoff):
```

Comment:
Just adding line breaks for readability
```suggestion

    for m in range(2, cutoff):
```